datafusion/core/src/physical_plan/file_format/mod.rs

```diff
@@ -1232,9 +1187,9 @@ mod tests {
             ],
         )
         .unwrap();
-
         let rows_num = batch.num_rows();
-        let mapped_batch = mapping.map_batch(batch).unwrap();
+        let projected = batch.project(&projection).unwrap();
```

Comment:
This highlights the major change, the schema adaptor assumes that the projection it output has been applied to the file_schema batches.